### ISSUE TITLE
grub-mkimage: Error: 

### ISSUE THREAD

**Issue by hyperclock** [emmett1/lfs-scripts#12]
grub-mkimage cannot find »/usr/lib/grub/x8664-efi/moddep.lst«

this is at the end of 03-mkiso

I noticed during part 02 the messages was about not being able to read some firmware, is this causing the probs?

**Comment by hyperclock** [comment]
It might be interesting to point out thar RedHat has a bug report of that with sot of an answer. It's at https://bugzilla.redhat.com/show_bug.cgi?id=1512493

**Comment by hyperclock** [comment]
I had to create  a symlink to get this running, acually 2 links.

```
ln -s /mnt/lfs/usr/lib/grub/x86_64-efi /usr/lib/grub/x86_64-efi
```
and

```
ln -s /mnt/lfs/usr/share/syslinux /usr/share/syslinux
```

**Comment by emmett1** [comment]
> I noticed during part 02 the messages was about not being able to read some firmware, is this causing the probs?

Yes, its fine to ignore, just warning from mkinitramfs script autodetecting firmware for kernel modules.

**Comment by emmett1** [comment]
> I had to create a symlink to get this running, acually 2 links.
> 
> ```
> ln -s /mnt/lfs/usr/lib/grub/x86_64-efi /usr/lib/grub/x86_64-efi
> ```
> 
> and
> 
> ```
> ln -s /mnt/lfs/usr/share/syslinux /usr/share/syslinux
> ```

It happen because its tried to use syslinux files from host, which might different between each distro. Thanks for report, i will fix it.

**Comment by hyperclock** [comment]
@emmett1 that did not fix it, I still need to symlink it. That maybe a Debian problem. I will try Ubuntu on the weekend, if that works I'll try to find out why and if it's appliable to Debian.

But symlinking does finish the iso.

**Comment by emmett1** [comment]
Please dont use ubuntu or any debian based distro. Ubuntu and debian basically the same thing. Debian often give problem when build lfs.

**Comment by hyperclock** [comment]
Sorry, but I'm a DDG (Debian Damaged Guy). Been using Debian exclusivly since 2005. Whenever a claim is made that Debian does not do something, it's my thing to find out why. 

It _does build_ these scripts and it does finish and _the iso is usable_. I will write a small short tutorial on that, hopefully by Friday evening or early Saturday.

The basics of doing this on Debian is stopping Dash, install some needed tools and software, run parts 01& 02, add 2 symlinks and run 03 - that's it.

**Comment by emmett1** [comment]
I did not claim debian bad or anything, but debian is too advanced they split packages badly until sometimes things missing can caused compiling stuff failed. If you could make it work on debian, i will add it to readme how to make it work.

**Comment by hyperclock** [comment]
I understood your claim, from a developers point of view. And I agree that Debian does do some things in a strange way. I was only claiming that with some effort, Debian does things a lot of people don't think it can. 

Anyway I'm in the process of setting up on of my "left over" domains and will add a small howto along with some screenshots of the way I use Virtualbox for this project.

**Comment by M0E-lnx** [comment]
So, I've hit this same error.

```:: Preparing kernel and initramfs...
Generating initramfs...
:: running build_hook: base
:: running build_hook: modules
WARNING: missing firmware for wd719x: wd719x-risc.bin
WARNING: missing firmware for wd719x: wd719x-wcs.bin
WARNING: missing firmware for aic94xx: aic94xx-seq.fw
:: running build_hook: udev
:: running build_hook: livecd
:: generating initramfs...
Generating initramfs done: /boot/initrd-live.img (24M)
:: Setup UEFI mode...
grub-mkimage: error: cannot open `/usr/lib/grub/x86_64-efi/moddep.lst': No such file or directory.
```
I'm not sure if this is looking in the host filesystem or in the built target, but I can confirm that the file does exist in the target that just got built.  Absolute path from the host is `/mnt/lfs/usr/lib/grub/x86_64-efi/moddep.lst` so if this is running in a chroot inside /mnt/lfs, this *should* work, but i'm not sure what the problem is really.

I should also mention that this file does not exist in my host filesystem (the debian 10 install used to build the LFS system)

**Comment by emmett1** [comment]
@M0E-lnx owh, looks like its trying to find grub-efi files from host. Try install ‘grub-efi’ (not sure if different distro called different name) to your host. I will try to make it find grub-efi files from target system.

**Comment by hyperclock** [comment]
@M0E-lnx that's why I symlinked those (there are 2 needed at this point) to the host  before running ```sudo ./03-mkiso```.

until now that was the only method I had to build thru 6 times.

@emmett1 I#m not sure about that. The scripts are looking for the grub-efi that was built during 02-base, symlinking these gives the script what it needs or is looking for, the script is just looking outside of the chroot somehow. I think if you add it to the host from it's package manager, then the script will get the info fron the host file and not those from the newly built LFS System.

**Comment by emmett1** [comment]
@hyperclock The 03-mkiso script is using tools from host to create iso, so it definitely use grub-efi from host except syslinux file i can point to newly built LFS System to use it. So nothing is wrong with the script. And it does not using chroot to create the iso. The only part script use chroot is to create initrd that support live boot. So to run 03-mkiso script, grub-efi is required for host system. Or maybe i will try to add it to temporary toolchain to use it.
